ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/components/properties/FontSpec.kt

```diff
@@ -110,29 +114,87 @@ internal fun FontSpec.resolve(
-            else -> getResourceIdentifier(name = info.value, type = "font")
-                .takeUnless { it == 0 }
-                ?.let { fontId -> FontSpec.Resource(id = fontId) }
-                ?: getAssetFontPath(name = info.value)
-                    ?.let { path -> FontSpec.Asset(path = path) }
-                ?: FontSpec.System(name = info.value).also {
-                    Logger.d(
-                        "Could not find a font resource named `${info.value}`. Assuming it's an OEM system font. " +
-                            "If it isn't, make sure the font exists in the `res/font` folder. See for more info: " +
-                            "https://developer.android.com/develop/ui/views/text-and-emoji/fonts-in-xml",
-                    )
-                }
+@Suppress("ReturnCount")
+private fun ResourceProvider.determineFontSpec(fontsConfig: FontsConfig): FontSpec {
+    val bundledFont = getBundledFontSpec(fontsConfig)
+    if (bundledFont != null) {
+        return bundledFont
+    }
+
+    val webFont = getDownloadedFontSpec(fontsConfig)
+    if (webFont != null) {
+        return webFont
+    }
+
+    return FontSpec.System(name = fontsConfig.android.value).also {
+        Logger.d(
+            "Could not find a font resource named `${fontsConfig.android.value}`. Assuming it's an OEM system font. " +
+                "If it isn't, make sure the font exists in the `res/font` folder. See for more info: " +
+                "https://developer.android.com/develop/ui/views/text-and-emoji/fonts-in-xml",
+        )
+    }
+}
+
+@Suppress("NestedBlockDepth")
+private fun ResourceProvider.getBundledFontSpec(
+    fontsConfig: FontsConfig,
+): FontSpec? {
+    val androidInfo = fontsConfig.android
+    val androidInfoValue = androidInfo.value
+    return if (androidInfoValue.isNotEmpty()) {
+        when (androidInfo) {
+            is FontInfo.GoogleFonts -> FontSpec.Google(name = androidInfoValue)
+            is FontInfo.Name -> when (androidInfoValue) {
+                FontFamily.SansSerif.name -> FontSpec.Generic.SansSerif
+                FontFamily.Serif.name -> FontSpec.Generic.Serif
+                FontFamily.Monospace.name -> FontSpec.Generic.Monospace
+                else -> getResourceIdentifier(name = androidInfoValue, type = "font")
+                    .takeIf { it != 0 }
+                    ?.let { fontId -> FontSpec.Resource(id = fontId) }
+                    ?: getAssetFontPath(name = androidInfoValue)
+                        ?.let { path -> FontSpec.Asset(path = path) }
+            }
         }
+    } else {
+        null
     }
+}
+
+private fun ResourceProvider.getDownloadedFontSpec(fontsConfig: FontsConfig): FontSpec.Downloaded? {
+    val webUrl = fontsConfig.web?.value?.takeIf { it.isNotEmpty() }
+    val webHash = fontsConfig.web?.hash?.takeIf { it.isNotEmpty() }
+    if (webUrl != null && webHash == null) {
+        Logger.w(
+            "Font `$webUrl` does not have a validation hash. " +
```

Comment:
Because the url is invisible in the dashboard, I would include the Android's font name in the log message, to make it easier for them to identify which font they should re-upload
```suggestion
            "Font `${fontsConfig.android.value}` does not have a validation hash. " +
```

Reply:
Good catch!